pkg/statistics/handle/usage/indexusage/collector.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexusage
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/parser/model"
+)
+
+// GlobalIndexID is the key type for indexUsageMap.
+type GlobalIndexID struct {
+	TableID int64
```

Comment:
If this table is a partitioning table, it seems like we'll record index usage for each partition individually, should we merge the same index data from different partitions together?

Reply:
I agree that the final display on `index_usage` table will only give one row for each table/index. 

However, I prefer to handle the aggregation of partition local indexes when we are retrieving the table, so that this package doesn't need to handle `infoschema` related things. Also, when we are retrieving the table, we'll need to filter out the table/index according to the session infoschema.